pennylane/devices/default_mixed.py

```diff
@@ -163,23 +163,10 @@ def density_matrix(self, wires):
             array[complex]: complex array of shape ``(2 ** len(wires), 2 ** len(wires))``
             representing the reduced density matrix of the state prior to measurement.
         """
-        # Return the full density matrix if all the wires are given
-        if wires == self.wires:
-            return self.state
+        wires = wires.tolist()
 
-        traced_wires = [x for x in self.wires if x not in wires]
-
-        # Trace first subsystem by applying kraus operators of the partial trace
-        tr_op = self._cast(np.eye(2), dtype=self.C_DTYPE)
-        tr_op = self._reshape(tr_op, (2, 1, 2))
-
-        self._apply_channel(tr_op, Wires(traced_wires[0]))
-
-        # Trace next subsystem by applying kraus operators of the partial trace
-        for traced_wire in traced_wires[1:]:
-            self._apply_channel(tr_op, Wires(traced_wire))
-
-        return self._reshape(self._state, (2 ** len(wires), 2 ** len(wires)))
+        state = np.reshape(self._pre_rotated_state, (2**self.num_wires, 2**self.num_wires))
+        return qnp.to_density_matrix(state, wires)
```

Comment:
Does this work for non-numeric wire labels? It's worth checking and maybe creating a separate test for it.

Reply:
It does not. `QubitDevice.statistics` raises an error *after* the entire execution has been done and we're just processing the results. Even though we could just apply the wire map to the custom labels and be fine. It's a problem.